edx_exams/apps/core/models.py

```diff
@@ -353,38 +355,69 @@ def create_or_update(cls, provider, course_id):
         provider_name = provider.name if provider else None
         existing_config = CourseExamConfiguration.get_configuration_for_course(course_id)
         if existing_config:
-            if existing_config.provider == provider:
-                # nothing to be done
-                log.info(f'Course exam configuration course_id={course_id} already has provider={provider_name}')
-                return
-            count = cls.update_course_config_provider(existing_config, provider)
-            log.info(f'Updated course exam configuration course_id={course_id} '
-                     + f'to provider={provider_name} and recreated {count} exams')
+            existing_provider = existing_config.provider
+
+            cls.update_course_config(existing_config, provider, escalation_email)
+
+            # If the provider is updated for a course, all existing exams have to be retired
+            # and duplicates made with the new provider.
+            if provider != existing_provider:
+                count = cls._sync_exams_with_new_provider(existing_config.course_id, provider)
+                log.info(f'Updated course exam configuration course_id={course_id} '
+                         + f'to provider={provider_name} and recreated {count} exams')
         else:
-            CourseExamConfiguration.objects.create(course_id=course_id, provider=provider)
-            log.info(f'Created course exam configuration course_id={course_id}, provider={provider_name}')
+            CourseExamConfiguration.objects.create(
+                course_id=course_id,
+                escalation_email=escalation_email,
+                provider=provider,
+            )
+            log.info(f'Created course exam configuration course_id={course_id}, provider={provider_name}, '
+                     + f'escalation_email={escalation_email}')
 
     @classmethod
-    def update_course_config_provider(cls, existing_config, new_provider):
+    def update_course_config(cls, existing_config, new_provider, escalation_email):
         """
-        If the provider is updated for a course, all existing exams have to be retired
-        and duplicates made with the new provider.
+        Update the provider and escalation_email fields of an instance of a CourseExamConfiguration model represented
+        by the existing_config parameter.
+
+        Parameters:
+        * existing_config: an instance of the CourseExamConfiguration model
+        * new_provider: an instance of the ProctoringProvider model; the provider to be set
+        * escalation_email: a string representing an email address; the escalation_email to be set
+        """
+        existing_config.provider = new_provider
+
+        if new_provider is None:
+            existing_config.escalation_email = ''
```

Comment:
I wanted to avoid a situation where someone sends this data `{'provider': None, 'escalation_email': 'test@example.com'}`. I believe we should always set the `escalation_email` to the empty string if there is no provider.

Reply:
oh this make sense. I misread the condition as `if escalation_email is None:` thinking we were doing some Null -> '' conversion.